machine/v1beta1/types_vsphereprovider.go

```diff
@@ -87,12 +87,124 @@ type NetworkSpec struct {
 	Devices []NetworkDeviceSpec `json:"devices"`
 }
 
+type DHCPState string
+
+const (
+	EnabledState  DHCPState = "Enabled"
+	DisabledState DHCPState = "Disabled"
+)
+
+// AddressesFromPool is an IPAddressPool that will be used to create
+// IPAddressClaims for fulfillment by an external controller.
+type AddressesFromPool struct {
+	Group string `json:"group"`
+
+	// Kind is the type of resource being referenced
+	Resource string `json:"resource"`
+
+	// Name is the name of resource being referenced
+	Name string `json:"name"`
+}
+
 // NetworkDeviceSpec defines the network configuration for a virtual machine's
 // network device.
 type NetworkDeviceSpec struct {
 	// NetworkName is the name of the vSphere network to which the device
 	// will be connected.
-	NetworkName string `json:"networkName"`
+	// +optional
+	NetworkName string `json:"networkName,omitempty"`
+
+	// DeviceName may be used to explicitly assign a name to the network device
```

Comment:
The godoc should start with the json tag version of the name, not the go version
```suggestion
	// deviceName may be used to explicitly assign a name to the network device
```